File: src/main/java/com/zippeykeys/praisebe/block/EOreBlock.java
package com.zippeykeys.praisebe.block;

import com.zippeykeys.praisebe.util.Localize;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
public enum EOreBlock implements Localize {
    // Normal //
    GALLIUM("gallium"), //
    PLATINUM("platinum"), //
    TITANIUM("titanium"), //
    TUNGSTEN("tungsten"), //
    CHROMIUM("chromium"), //

    // Enchanted //
    GALVORN("galvorn"), //
    MITHRIL("mithril"), //
    ORICHALCUM("orichalcum"), //
    ADAMANTITE("adamantite"), //
    SCARLETITE("scarletite");

    @Getter(onMethod_ = @Override)
    private String name;

    @Override
    public String getPrefix() {
        return "building";
    }
}

File: src/main/kotlin/com/zippeykeys/praisebe/block/EOreBlock.kt
package com.zippeykeys.praisebe.block

import com.zippeykeys.praisebe.iface.ILocalize

enum class EOreBlock(private val actualName: String) : ILocalize {
    // Normal //
    GALLIUM("gallium"),
    PLATINUM("platinum"),
    TITANIUM("titanium"),
    TUNGSTEN("tungsten"),
    CHROMIUM("chromium"),
    // Enchanted //
    GALVORN("galvorn"),
    MITHRIL("mithril"),
    ORICHALCUM("orichalcum"),
    ADAMANTITE("adamantite"),
    SCARLETITE("scarletite");

    override val prefix: String
        get() = "building"

    override fun getName(): String {
        return actualName
    }
}